view/frontend/web/js/track-checkout.js

```diff
@@ -0,0 +1,72 @@
+/* Track checkout funnel transition in magento's default checkout */
+
+// TODO: support logged in user
+
+var currentStep = '';
+
+function waitFor(condition, doWhenReady, pollIntervalMillis) {
+    var defaultPollIntervalMillis = 100;
+    var maxRetryCount = 30;
+    var checkCondition = function (retryCount) {
```

Comment:
Not really functionally much different. Just might be a little cleaner to do something like this but I'm find without it.

> function waitFor(condition, doWhenReady, pollIntervalMillis = 100, maxRetryCount = 30) {    
>     var checkCondition = function (retryCount) {
>         if (condition()) {
>             doWhenReady();
>         } else if (retryCount < maxRetryCount) {
>             setTimeout(function(){checkCondition(retryCount + 1)}, pollIntervalMillis);
>         }
>     };
>     checkCondition(0);
> }

Reply:
unfortunately IE doesn't support default argument value